### PR TITLE
Haiku fixes

### PR DESCRIPTION
**Pull request by kallisti5**
Some basic fixes to get nheko compiling under Haiku's Qt libraries

### CODE REVIEW

**Comment by kallisti5** [comment]
receipts:
```
[2021-02-03 14:36:11.193] [net] [info] initial sync completed
[2021-02-03 14:36:11.211] [db] [debug] saving '{"event_id":"$iZAaU-JDtYX_nyQZzqEZ3tvKo5y2xCq7ywFDnHMEt6o","prev_batch":"t7965-1792105308_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.211] [db] [debug] saving '{"event_id":"$kHUBhzKjFVb25fF42dc0b7YzBySZthMTHwZEDBdElZM"}'
[2021-02-03 14:36:11.212] [db] [debug] saving '{"event_id":"$Jxw57P1YOZORtGnoxMUKQrTjuz7Tb_IF8vcyky4GFE0"}'
[2021-02-03 14:36:11.212] [db] [debug] saving '{"event_id":"$LU5PQUaFdV6w2WKTJ3UoLLfF0pO80q5F_8w5BEUklxs"}'
[2021-02-03 14:36:11.212] [db] [debug] saving '{"event_id":"$rx0OJujcsPL7XPLTTXXkNzccJqyJ-I9gMJBvPXth578"}'
[2021-02-03 14:36:11.212] [db] [debug] saving '{"event_id":"$7GhIhmL5hjt5Gj_88ni_Gu4RgGGZrvCnqsHXr51rVBk"}'
[2021-02-03 14:36:11.212] [db] [debug] saving '{"event_id":"$oP2TBlvZvhv-1uqeRBgvQDnPrPIA8Mea_qH-JRs_QxM"}'
[2021-02-03 14:36:11.212] [db] [debug] saving '{"event_id":"$G70hW6pERBKi5Bt2J4n2Kv7PkNYaMP8wAeIb0aRHEDk"}'
[2021-02-03 14:36:11.212] [db] [debug] saving '{"event_id":"$gO8xShK4TiDGfctaEygyo-1ytJ-p4C-HVZg9eM9Id88"}'
[2021-02-03 14:36:11.213] [db] [debug] saving '{"event_id":"$rtEAhhQkgOFd2vEHjvrNmroQn918qLYTgH1VsoTP1Is"}'
[2021-02-03 14:36:11.213] [db] [debug] Fully read: $C6vWh20mot5K6xKlx1ayiFnxXUr3FAz7KWN97tTypf8
[2021-02-03 14:36:11.245] [db] [debug] saving '{"event_id":"$SJ3--uK5Q_dm2i-1F3j2zeFYcjV99fPXkOjsiEj9MTY","prev_batch":"t89-1773448036_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.245] [db] [debug] saving '{"event_id":"$bfr9d1QtapDaE_1h9MTO_fTyOknuQfmdA9st9OdJ10s"}'
[2021-02-03 14:36:11.246] [db] [debug] saving '{"event_id":"$GLbdsPgDOH0_2QGdSU--TvieLVIhTz8zW6cQgbdtoDw"}'
[2021-02-03 14:36:11.246] [db] [debug] saving '{"event_id":"$-YjzRFc-YLkQtTi6L963VN2oMyBLota0DJGUAGzCXxc"}'
[2021-02-03 14:36:11.246] [db] [debug] saving '{"event_id":"$93En4k6BYnIPnjAUYgwC2yHPSPF5XM39cqdukkhx0Fc"}'
[2021-02-03 14:36:11.246] [db] [debug] saving '{"event_id":"$_5QPq97B9VuW_aiBtcb1fc1RIw4o-gPcTJjLO2w6j28"}'
[2021-02-03 14:36:11.246] [db] [debug] saving '{"event_id":"$Mp1KxwKrizeFcHeZBInqxJH3-AHiVkIb4oVukpLzZLk"}'
[2021-02-03 14:36:11.246] [db] [debug] saving '{"event_id":"$USBY0FJep6KhzQ4Y9RqieCwacUkp59kC2KmyoA-B4WA"}'
[2021-02-03 14:36:11.246] [db] [debug] saving '{"event_id":"$LTSk9KmnDx1u8T2GFpGTCniUS908vToKZCPW3KxHnSY"}'
[2021-02-03 14:36:11.247] [db] [debug] saving '{"event_id":"$IvPMQOIxb0k90dv0QcLI_I4Z9qabQzUhLcoKK1z75ck"}'
[2021-02-03 14:36:11.248] [db] [debug] Fully read: $USBY0FJep6KhzQ4Y9RqieCwacUkp59kC2KmyoA-B4WA
[2021-02-03 14:36:11.249] [db] [debug] saving '{"event_id":"$158639704061691qdwdi:matrix.org","prev_batch":"t23-1331515166_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.249] [db] [debug] saving '{"event_id":"$1594145120503614AqPPh:matrix.org"}'
[2021-02-03 14:36:11.250] [db] [debug] saving '{"event_id":"$15941586455777qNRXK:matrix.org"}'
[2021-02-03 14:36:11.250] [db] [debug] saving '{"event_id":"$159492198113910IAVaP:matrix.org"}'
[2021-02-03 14:36:11.250] [db] [debug] saving '{"event_id":"$159492201013954UKRtb:matrix.org"}'
[2021-02-03 14:36:11.250] [db] [debug] saving '{"event_id":"$159492204614019qEeKs:matrix.org"}'
[2021-02-03 14:36:11.250] [db] [debug] saving '{"event_id":"$159492206614062GzEuW:matrix.org"}'
[2021-02-03 14:36:11.250] [db] [debug] saving '{"event_id":"$159492369016598IRGpq:matrix.org"}'
[2021-02-03 14:36:11.250] [db] [debug] saving '{"event_id":"$1596487912428hNNKq:das-labor.org"}'
[2021-02-03 14:36:11.251] [db] [debug] saving '{"event_id":"$1601300962494883gtJDW:matrix.org"}'
[2021-02-03 14:36:11.251] [db] [debug] Fully read: $159492369016598IRGpq:matrix.org
[2021-02-03 14:36:11.253] [db] [debug] saving '{"event_id":"$a8MV7Tt7JHQTw4V7osnSgR5kBSEjQPs1zFEaPQN69qY","prev_batch":"t6-1459548784_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.253] [db] [debug] saving '{"event_id":"$avSqp90Tehm9ME1j1qqKn5CkqpYG1v3wQxcrCylEjSU"}'
[2021-02-03 14:36:11.253] [db] [debug] saving '{"event_id":"$4vbiSDqAQRCfRMTfc39Z4OxcCI5IABxate43jYjUFRw"}'
[2021-02-03 14:36:11.253] [db] [debug] saving '{"event_id":"$pvRRNivandbvkSTxc_XMi5AC2dj3hZf_J7BnDXQsP10"}'
[2021-02-03 14:36:11.253] [db] [debug] saving '{"event_id":"$sSzrft663R_mftgz5z9PJOMDcKjaIi4Cn59gaiusMZ4"}'
[2021-02-03 14:36:11.254] [db] [debug] saving '{"event_id":"$l6Tpi5AapAvv2pc0ILuMYGlpmmbHUcqZE40hQO1rPWI"}'
[2021-02-03 14:36:11.254] [db] [debug] saving '{"event_id":"$TmfujoDL2INA6b1CZuj4cE8ax_0w_1F96jjouxf5Kk4"}'
[2021-02-03 14:36:11.254] [db] [debug] saving '{"event_id":"$D-RI_DTAHwl4wxFIOm1yB-YSep4x_aeUT_RFmoZ4JoQ"}'
[2021-02-03 14:36:11.254] [db] [debug] saving '{"event_id":"$I26Sl-SPwM_YZH5dujHfiUcv4vYOC4CueCjtxtUwYIs"}'
[2021-02-03 14:36:11.254] [db] [debug] saving '{"event_id":"$3a1kfc4TpX3QzorbNsVqDSbkVL6L9AHgV9rmCQh3zY8"}'
[2021-02-03 14:36:11.255] [db] [debug] Fully read: $3a1kfc4TpX3QzorbNsVqDSbkVL6L9AHgV9rmCQh3zY8
[2021-02-03 14:36:11.263] [db] [debug] saving '{"event_id":"$tO5EC7HKCG0rITXXkbF7EzRQibzMGXrqr8jcVQw4S-c","prev_batch":"t1258-1788434347_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.264] [db] [debug] saving '{"event_id":"$ON0hUyD7wc4FRkH6HmOPhxCU63ThWGxQCKu5h38ITtM"}'
[2021-02-03 14:36:11.264] [db] [debug] saving '{"event_id":"$yIVU5Kw_q5Kw0ryB4ZvdX9g6efAT7B0mrnlvmwaUqmA"}'
[2021-02-03 14:36:11.265] [db] [debug] saving '{"event_id":"$ixKEJx8WCfFyQ0-zNv_B35C-vdIFcQGxk0LuLYgG5gw"}'
[2021-02-03 14:36:11.265] [db] [debug] saving '{"event_id":"$eAeC7K4R-3SKtzWxhq3jBgN3a6PMS_IfxaaJNxxEN04"}'
[2021-02-03 14:36:11.265] [db] [debug] saving '{"event_id":"$erDZiSYLE43WZNzB2h45BcG13B69prEvO7SCJ2RTrBY"}'
[2021-02-03 14:36:11.265] [db] [debug] saving '{"event_id":"$134mw41Uwm_882lkQmeHtJBl1kOVAjStN3-wWAlgYa0"}'
[2021-02-03 14:36:11.265] [db] [debug] saving '{"event_id":"$rXwuf0wHi6oO0QqjW3MpdAhHTJDpj0HUdEC-HamuS_U"}'
[2021-02-03 14:36:11.265] [db] [debug] saving '{"event_id":"$saDnFViJHqTuL4RDVJBOR6eD0iFRcOi66tfKRUphpJc"}'
[2021-02-03 14:36:11.266] [db] [debug] saving '{"event_id":"$Xc9uFwTntRDVAvGK_TMt9IslQX_CLxU8euE5WJieG3Y"}'
[2021-02-03 14:36:11.266] [db] [debug] Fully read: $yIVU5Kw_q5Kw0ryB4ZvdX9g6efAT7B0mrnlvmwaUqmA
[2021-02-03 14:36:11.292] [db] [debug] saving '{"event_id":"$dYLKvsHEH_UgDheIt5OkHIlv6GTjoTjs82GWVdTUa0s","prev_batch":"t9350-1789137212_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.292] [db] [debug] saving '{"event_id":"$KVYfpZ_JA7PX6KZmBdxs27hTIU8j_f8Y4iC37_gklJM"}'
[2021-02-03 14:36:11.292] [db] [debug] saving '{"event_id":"$5g1qwE71M3tC9d9G1fZIHFXRgtV6XB8Aq9KOnuX8aVc"}'
[2021-02-03 14:36:11.293] [db] [debug] saving '{"event_id":"$e6_Of7QeYPbIL1XLDc9V4VlrX9iqUyG7YgazKmBDxUI"}'
[2021-02-03 14:36:11.293] [db] [debug] saving '{"event_id":"$Iwe-fk8auzLzEzOZT1Hw6ub2YPNw7teH86QdfQt1nns"}'
[2021-02-03 14:36:11.293] [db] [debug] saving '{"event_id":"$TCoik3qbUrQsJYncgve8dTM4vn4D026nAQzjMrUhEXE"}'
[2021-02-03 14:36:11.293] [db] [debug] saving '{"event_id":"$ECRKtTSSMXlfseUvXJSjmK0nTSUtiDaX_mRP2NivbUY"}'
[2021-02-03 14:36:11.293] [db] [debug] saving '{"event_id":"$OTRJWvq2wmBHIrmIR5gqYegfdJJFimefBVb5zjjUHzA"}'
[2021-02-03 14:36:11.293] [db] [debug] saving '{"event_id":"$CO_-aVRaInvgDHKV4BCL38ioWp1fnADDxRoFt0jvyD4"}'
[2021-02-03 14:36:11.294] [db] [debug] saving '{"event_id":"$HzeS9P9XnwvqfoFdO-0O-gVtx47yP-BZ5WLPQHMblEQ"}'
[2021-02-03 14:36:11.294] [db] [debug] Fully read: $QWwfYIuNJ6CaXvqc2-ST7Asm-6Tc-YfypFneIVVo2B0
[2021-02-03 14:36:11.409] [db] [debug] saving '{"event_id":"$161236898625680HxtTn:matrix.org","prev_batch":"t19867-1791979393_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.409] [db] [debug] saving '{"event_id":"$1612370489654HBeNJ:matrix.org"}'
[2021-02-03 14:36:11.409] [db] [debug] saving '{"event_id":"$16123736483442ayDDB:matrix.org"}'
[2021-02-03 14:36:11.409] [db] [debug] saving '{"event_id":"$16123740273779hNjgq:matrix.org"}'
[2021-02-03 14:36:11.409] [db] [debug] saving '{"event_id":"$16123743854144ARKqc:matrix.org"}'
[2021-02-03 14:36:11.409] [db] [debug] saving '{"event_id":"$16123753935034iQCkn:matrix.org"}'
[2021-02-03 14:36:11.410] [db] [debug] saving '{"event_id":"$16123778206940sGfDh:matrix.org"}'
[2021-02-03 14:36:11.410] [db] [debug] saving '{"event_id":"$16123792168149vBdTQ:matrix.org"}'
[2021-02-03 14:36:11.410] [db] [debug] saving '{"event_id":"$161238334611419nyGQw:matrix.org"}'
[2021-02-03 14:36:11.410] [db] [debug] saving '{"event_id":"$161238338111445jBRHp:matrix.org"}'
[2021-02-03 14:36:11.411] [db] [debug] Fully read: $1612273926493234aeqvA:matrix.org
[2021-02-03 14:36:11.444] [db] [debug] saving '{"event_id":"$_58FQwhuoLJyM8TzyfXGWqOpPgX7ai5AIR062-CZbI4","prev_batch":"t1124-1771093091_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.444] [db] [debug] saving '{"event_id":"$RU4bKVZSrqGpIBF_xCEt-Dp2ZYDOk_7ivoDF54hYG1I"}'
[2021-02-03 14:36:11.444] [db] [debug] saving '{"event_id":"$k0wUYNdG3KYdsKYXfcPETlnhAstrMe6nL0PJO_QNcoQ"}'
[2021-02-03 14:36:11.444] [db] [debug] saving '{"event_id":"$9oB5GpBmSiNcoz6P4sk2If9CFHtHoaJkMGqkH1r6poo"}'
[2021-02-03 14:36:11.444] [db] [debug] saving '{"event_id":"$20-B-62LQbTwIv3d28M6RXxn0cu6DtPhSMjfc087Q_g"}'
[2021-02-03 14:36:11.444] [db] [debug] saving '{"event_id":"$1jFDeoLieVdlpukyt4_s_KZAsUdKDskNTjrrbNxtCQg"}'
[2021-02-03 14:36:11.445] [db] [debug] saving '{"event_id":"$tk8FLc6fgRE--dNe43shAtYv89Qydu0IE0b-pit-3sg"}'
[2021-02-03 14:36:11.445] [db] [debug] saving '{"event_id":"$sF8GBQANP-4Du7afZihmqYKokjhgaXJaQtBS-kpD2VM"}'
[2021-02-03 14:36:11.445] [db] [debug] saving '{"event_id":"$KuN6Oi4luQqdDisviL0ms7RCOv7bea8lDuKSFdbmseo"}'
[2021-02-03 14:36:11.445] [db] [debug] saving '{"event_id":"$LrrsrKSNFUfWGoHhT1Hrv1ALYxQZoj7daK0URQ0OlqY"}'
[2021-02-03 14:36:11.446] [db] [debug] Fully read: $LrrsrKSNFUfWGoHhT1Hrv1ALYxQZoj7daK0URQ0OlqY
[2021-02-03 14:36:11.460] [db] [debug] saving '{"event_id":"$kD19VHeQA8BRCCWEXtGRjLsm4CVt11WAqZ4lz5S9Wrk","prev_batch":"t7693-1790107652_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.460] [db] [debug] saving '{"event_id":"$zPkzCiJx5M0nSTGFCNHiRF9hc5vTCJTjk9qYZa78pNA"}'
[2021-02-03 14:36:11.460] [db] [debug] saving '{"event_id":"$TQOPOpGv5Lth61Uqo6o_P3R6ugW56IKusd-urK-swmQ"}'
[2021-02-03 14:36:11.460] [db] [debug] saving '{"event_id":"$u3qj8DZSrEHE3qjXdjITEfL-dWzBCxnizQGJKQ1kRM0"}'
[2021-02-03 14:36:11.460] [db] [debug] saving '{"event_id":"$PmkKlMSyGVK-cLGpgwhnbO7cUWAFOGyMorH9PbXV-tA"}'
[2021-02-03 14:36:11.461] [db] [debug] saving '{"event_id":"$68abXvsqJyqadHtras_OWgJQ601G4XHja2jYmQ5Aj7E"}'
[2021-02-03 14:36:11.461] [db] [debug] saving '{"event_id":"$FbQtgoba-Nb0KVBesPjMmpLdWDtYumVi0v7VbxEMpws"}'
[2021-02-03 14:36:11.461] [db] [debug] saving '{"event_id":"$1LcnGVmF79wcgv9H8dR9yCWDEKFe3lJN13JWQAZreEY"}'
[2021-02-03 14:36:11.461] [db] [debug] saving '{"event_id":"$JpeGDbcibYFRvichta0e-e4VePD9gXJUDCY3SnQL5O8"}'
[2021-02-03 14:36:11.461] [db] [debug] saving '{"event_id":"$Yy10u4ofcI2pNAfFbgkmO8oxnf2Qhg9-JejB9t5gpJk"}'
[2021-02-03 14:36:11.462] [db] [debug] Fully read: $kD19VHeQA8BRCCWEXtGRjLsm4CVt11WAqZ4lz5S9Wrk
[2021-02-03 14:36:11.485] [db] [debug] saving '{"event_id":"$1612318117114967nejUQ:matrix.org","prev_batch":"t36930-1790827480_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.485] [db] [debug] saving '{"event_id":"$1612318678116237auvzx:matrix.org"}'
[2021-02-03 14:36:11.485] [db] [debug] saving '{"event_id":"$1612326940132619vMjVN:matrix.org"}'
[2021-02-03 14:36:11.486] [db] [debug] saving '{"event_id":"$1612334615145411tcMDk:matrix.org"}'
[2021-02-03 14:36:11.486] [db] [debug] saving '{"event_id":"$1612335179146256cLnOn:matrix.org"}'
[2021-02-03 14:36:11.486] [db] [debug] saving '{"event_id":"$1612338067151429UBMbv:matrix.org"}'
[2021-02-03 14:36:11.486] [db] [debug] saving '{"event_id":"$1612338636152655MODiR:matrix.org"}'
[2021-02-03 14:36:11.486] [db] [debug] saving '{"event_id":"$1612340324156535SYIjS:matrix.org"}'
[2021-02-03 14:36:11.486] [db] [debug] saving '{"event_id":"$161236871772364KbIgQ:matrix.org"}'
[2021-02-03 14:36:11.487] [db] [debug] saving '{"event_id":"$161238036636170MwEvX:matrix.org"}'
[2021-02-03 14:36:11.487] [db] [debug] Fully read: $16122788991368522YCBnJ:matrix.org
[2021-02-03 14:36:11.490] [db] [debug] saving '{"event_id":"$6N2nEXbyrrlsFIeJF-tkuoJMI3M85Af41cWGMQglZVw","prev_batch":"t4-1777924762_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.491] [db] [debug] saving '{"event_id":"$oh2Ca4Uuv6_Yqkk31nd6_79137Ht98chbFPNJ3prG6I"}'
[2021-02-03 14:36:11.491] [db] [debug] saving '{"event_id":"$9Gj4rxo4o7aHVYBUJ2auJTuBhR80YnA7vTAVmjqY1Co"}'
[2021-02-03 14:36:11.491] [db] [debug] saving '{"event_id":"$lIbXWZj1R3c_55eHlXFFHnvfUwaSaXPvmgWeEGa6Yk0"}'
[2021-02-03 14:36:11.491] [db] [debug] saving '{"event_id":"$KICJVgDY2h8KjVkaVnhOX10GkcjYAgsLiVV_mGf28GE"}'
[2021-02-03 14:36:11.491] [db] [debug] saving '{"event_id":"$7G7BiZm63-MO0_Po7Gnkmshf4hI1oILudAzjWs94MOI"}'
[2021-02-03 14:36:11.491] [db] [debug] saving '{"event_id":"$7BpY6JxKWk5qq3U_Fkd1EAmIIOEcUEQevfp7K7BC6kQ"}'
[2021-02-03 14:36:11.491] [db] [debug] saving '{"event_id":"$jQMAGCluPM5JMAk-yR-bn48XWNL08gEglaIguLwHGxY"}'
[2021-02-03 14:36:11.492] [db] [debug] saving '{"event_id":"$KD2buKp7tpReB4ARTvraCSX2JpH85nmtKt28rgWifbo"}'
[2021-02-03 14:36:11.492] [db] [debug] saving '{"event_id":"$cCVOIsQFHxLu9aFtTL64P31ObTOwsuf8z7wzqijj9OM"}'
[2021-02-03 14:36:11.492] [db] [debug] Fully read: $cCVOIsQFHxLu9aFtTL64P31ObTOwsuf8z7wzqijj9OM
[2021-02-03 14:36:11.494] [db] [debug] saving '{"event_id":"$1609382296134093xEano:matrix.org","prev_batch":"t53-1723126147_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.494] [db] [debug] saving '{"event_id":"$1609382804134726YrJPW:matrix.org"}'
[2021-02-03 14:36:11.494] [db] [debug] saving '{"event_id":"$1609598834349657udMAW:matrix.org"}'
[2021-02-03 14:36:11.494] [db] [debug] saving '{"event_id":"$1609635803387534FSxoq:matrix.org"}'
[2021-02-03 14:36:11.494] [db] [debug] saving '{"event_id":"$1609635976387635oolOe:matrix.org"}'
[2021-02-03 14:36:11.495] [db] [debug] saving '{"event_id":"$16102992150tdPVC:talk.matthewgall.chat"}'
[2021-02-03 14:36:11.495] [db] [debug] saving '{"event_id":"$161133896867898hPUGV:matrix.org"}'
[2021-02-03 14:36:11.495] [db] [debug] saving '{"event_id":"$16120107064QjStB:nitro.chat"}'
[2021-02-03 14:36:11.495] [db] [debug] saving '{"event_id":"$161207101489383ICmmb:matrix.org"}'
[2021-02-03 14:36:11.495] [db] [debug] saving '{"event_id":"$1612162941141025qLlBA:matrix.org"}'
[2021-02-03 14:36:11.496] [db] [debug] Fully read: $1609635976387635oolOe:matrix.org
[2021-02-03 14:36:11.497] [db] [info] mark room !yKRpYGwtwGoQOylObd:matrix.org as encrypted
[2021-02-03 14:36:11.498] [db] [debug] saving '{"event_id":"$m2ZqEbFOnP5GZvWZAGT0uva0jb_jXtvLK1P_ZW182xc","prev_batch":"t3-1460367437_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.498] [db] [debug] saving '{"event_id":"$4E4w3CQZteg5fPqpfSc6FOypYiCviIIJsHyDQhgoi38"}'
[2021-02-03 14:36:11.498] [db] [debug] saving '{"event_id":"$9Vvsd1h1doVoziDzXKN-4xAxIE6MqLiOr96XUEwhCpY"}'
[2021-02-03 14:36:11.498] [db] [debug] saving '{"event_id":"$Nu7Sli53oYfsA_hatTehQ3RU0_j42L9e5Y-E5ZGpAPA"}'
[2021-02-03 14:36:11.498] [db] [debug] saving '{"event_id":"$0FdMUk2mEbnDSmdAt5o1f8fC_NzKDOGilhNYaxd6frE"}'
[2021-02-03 14:36:11.498] [db] [debug] saving '{"event_id":"$l-ZWVHd5_ubHib9AWekmYNpEzxrmf-9pZDkho9rF1MM"}'
[2021-02-03 14:36:11.498] [db] [debug] saving '{"event_id":"$BjW0aAagIYN-NhIRC5_RNMmPiUOGkH5aB-CnPIiZ4kc"}'
[2021-02-03 14:36:11.499] [db] [debug] saving '{"event_id":"$B_t_1qRg4L4L3g8FMdm4xU77LqS4yFKkI08_-SSoxE0"}'
[2021-02-03 14:36:11.499] [db] [debug] saving '{"event_id":"$RAqUP914zfB8bvxL5Vn31PcWyMZbvR88kprOMdYQFuY"}'
[2021-02-03 14:36:11.499] [db] [debug] saving '{"event_id":"$TgrX-2lYWod_FYF34T7ZsZRywH-X9xVdAwkvFKWalmc"}'
[2021-02-03 14:36:11.500] [db] [debug] Fully read: $TgrX-2lYWod_FYF34T7ZsZRywH-X9xVdAwkvFKWalmc
[2021-02-03 14:36:11.610] [db] [debug] saving '{"event_id":"$161237554920479XoMiB:matrix.org","prev_batch":"t465019-1792209062_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.610] [db] [debug] saving '{"event_id":"$161237555020482bfFPG:matrix.org"}'
[2021-02-03 14:36:11.610] [db] [debug] saving '{"event_id":"$16123782797544gcLjt:matrix.org"}'
[2021-02-03 14:36:11.610] [db] [debug] saving '{"event_id":"$161237849729lmFiY:halogen.city"}'
[2021-02-03 14:36:11.611] [db] [debug] saving '{"event_id":"$161237852730iZZxs:halogen.city"}'
[2021-02-03 14:36:11.611] [db] [debug] saving '{"event_id":"$16123800669227eicqQ:matrix.org"}'
[2021-02-03 14:36:11.611] [db] [debug] saving '{"event_id":"$16123816161543vMkKD:privacytools.io"}'
[2021-02-03 14:36:11.611] [db] [debug] saving '{"event_id":"$16123813950ibIGL:bitsbyt.es"}'
[2021-02-03 14:36:11.611] [db] [debug] saving '{"event_id":"$16123821412PwQMi:bitsbyt.es"}'
[2021-02-03 14:36:11.611] [db] [debug] saving '{"event_id":"$16123821513Lfqgc:bitsbyt.es"}'
[2021-02-03 14:36:11.612] [db] [debug] Fully read: $16122899437514FyilB:matrix.org
[2021-02-03 14:36:11.614] [db] [debug] saving '{"event_id":"$1606889569456497WsqkX:matrix.org","prev_batch":"t70-1674095123_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.614] [db] [debug] saving '{"event_id":"$1606889581456516JcePF:matrix.org"}'
[2021-02-03 14:36:11.614] [db] [debug] saving '{"event_id":"$1606889581456517CTqYf:matrix.org"}'
[2021-02-03 14:36:11.614] [db] [debug] saving '{"event_id":"$1608756433754308LwkEL:matrix.org"}'
[2021-02-03 14:36:11.614] [db] [debug] saving '{"event_id":"$161015338367369ncaSl:matrix.org"}'
[2021-02-03 14:36:11.614] [db] [debug] saving '{"event_id":"$161015340867409GDMje:matrix.org"}'
[2021-02-03 14:36:11.614] [db] [debug] saving '{"event_id":"$161015574970045UqQEK:matrix.org"}'
[2021-02-03 14:36:11.615] [db] [debug] saving '{"event_id":"$16102977156835qQaDW:librem.one"}'
[2021-02-03 14:36:11.615] [db] [debug] saving '{"event_id":"$161141875687871XktXW:matrix.org"}'
[2021-02-03 14:36:11.615] [db] [debug] saving '{"event_id":"$16116768644358SufOQ:matrix.org"}'
[2021-02-03 14:36:11.616] [db] [debug] Fully read: $161015574970045UqQEK:matrix.org
[2021-02-03 14:36:11.825] [db] [debug] saving '{"event_id":"$161238159110019nEThC:matrix.org","prev_batch":"t115417-1792422720_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533"}'
[2021-02-03 14:36:11.825] [db] [debug] saving '{"event_id":"$161238168710181gbjtb:matrix.org"}'
[2021-02-03 14:36:11.825] [db] [debug] saving '{"event_id":"$161238174228yXRLj:matrix.org"}'
[2021-02-03 14:36:11.826] [db] [debug] saving '{"event_id":"$161238176931pJIeZ:matrix.org"}'
[2021-02-03 14:36:11.826] [db] [debug] saving '{"event_id":"$161238248610801tmywo:matrix.org"}'
[2021-02-03 14:36:11.826] [db] [debug] saving '{"event_id":"$161238255410869eCHzA:matrix.org"}'
[2021-02-03 14:36:11.826] [db] [debug] saving '{"event_id":"$16123834044809LSOKr:in.tum.de"}'
[2021-02-03 14:36:11.826] [db] [debug] saving '{"event_id":"$161238371311767kHlcR:matrix.org"}'
[2021-02-03 14:36:11.826] [db] [debug] saving '{"event_id":"$161238387911939sojGV:matrix.org"}'
[2021-02-03 14:36:11.827] [db] [debug] saving '{"event_id":"$161238450312675qYRbJ:matrix.org"}'
[2021-02-03 14:36:11.827] [db] [debug] Fully read: $161178609217tFADQ:matrix.shadow53.com
[2021-02-03 14:36:12.488] [ui] [warning] description update on non-existent room_id: !InCYrriOILVfcNhfyZ:jupiterbroadcasting.com, carlwgeorge: https://www.redhat.com/en/blog/what-open-source-upstream
[2021-02-03 14:36:12.489] [ui] [warning] description update on non-existent room_id: !MHIHSBmgcvTtQPouOH:matrix.org, Yannick Barbel: https://github.com/Nheko-Reborn/nheko
[2021-02-03 14:36:12.490] [ui] [warning] description update on non-existent room_id: !MattiYOZpAFKFNFKic:matrix.org, You joined this room.
[2021-02-03 14:36:12.491] [ui] [warning] description update on non-existent room_id: !QXtjqcuxdAtsrXvfwS:matrix.org, You joined this room.
[2021-02-03 14:36:12.492] [ui] [warning] description update on non-existent room_id: !UMmuqvidWBzKGimLkt:matrix.org, Y (´･_･`): :)
[2021-02-03 14:36:12.496] [ui] [warning] description update on non-existent room_id: !WMZPgpQpdMulhwFOdV:matrix.org, ossalad: Has anyone heard of algorand?
[2021-02-03 14:36:12.497] [ui] [warning] description update on non-existent room_id: !bSyzQiSVSnbpgIBXdY:jupiterbroadcasting.com, AtypicalKernel: Podman coming down the pipe can even use docker commands
[2021-02-03 14:36:12.498] [ui] [warning] description update on non-existent room_id: !bUUVbPAXYHHAWqLRkc:matrix.org, K70S: hopefully you understand my question
[2021-02-03 14:36:12.502] [ui] [warning] description update on non-existent room_id: !fWeGuYFJOGagYrCdaY:matrix.org, @oftc-irc:matrix.org sent a notification
[2021-02-03 14:36:12.504] [ui] [warning] description update on non-existent room_id: !vLlJLzdhxrZxaWmrBH:matrix.org, You joined this room.
[2021-02-03 14:36:12.505] [ui] [warning] description update on non-existent room_id: !yKRpYGwtwGoQOylObd:matrix.org, You joined this room.
[2021-02-03 14:36:12.507] [qml] [warning] QObject::connect: Cannot queue arguments of type 'QMap<QString,mtx::responses::Notifications>'
(Make sure 'QMap<QString,mtx::responses::Notifications>' is registered using qRegisterMetaType().) (:0, )
[2021-02-03 14:36:12.509] [ui] [warning] description update on non-existent room_id: !yhqiEdqNjyPbxtUjzm:matrix.org, octav1a: computer?
[2021-02-03 14:36:12.511] [ui] [warning] description update on non-existent room_id: !zOPQkWMsZzrdiVQJym:matrix.org, psydroid: Hello
[2021-02-03 14:36:12.740] [ui] [info] initialize room list
[2021-02-03 14:36:12.747] [ui] [info] Trying to activate room !bUUVbPAXYHHAWqLRkc:matrix.org
[2021-02-03 14:36:12.865] [ui] [info] Activated room !bUUVbPAXYHHAWqLRkc:matrix.org
[2021-02-03 14:36:12.905] [ui] [debug] Paginating room !bUUVbPAXYHHAWqLRkc:matrix.org, token t7693-1790107652_757284961_730486_727126482_590497628_2003748_202005711_693553281_166533
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -645,7 +645,7 @@ if(QML_DEBUGGING)
 endif()
 
 
-if(NOT MSVC)
+if(NOT MSVC AND NOT HAIKU)
```
Comment (deepbluev7) [inline CMakeLists.txt:648]: Hm, -Werror is only used in debug builds, which should not affect release or release with debug builds. I would be interested in what warnings there are on Haiku, that aren't on other platforms? Maybe we can fix them instead?

File: src/notifications/Manager.h [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #include <QObject>
 #include <QString>
 
-#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_HAIKU)
```
Comment (deepbluev7) [inline src/notifications/Manager.h:7]: Oh, neat, Haiku supports dbus for notifications as well?

**Review by deepbluev7 — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (deepbluev7) [reply CMakeLists.txt:648]: Well, I guess there is no reason to hold this up from getting merged, just to wait for a reply :D

**Comment by deepbluev7** [comment]
Thank you for doing the porting work, very much appreciated!

**Pull request merged**